### DISCUSSION TITLE
Precursor Counts

### DISCUSSION THREAD

**Discussion by abadgerw** [vdemichev/DiaNN#1223] [Q&A] [answered]
Thank you for such an amazing tool. I am using the unique genes matrix as part of my analysis. However, I wanted to have the corresponding precursor counts for each gene in that matrix. Any insight into which files I can utilize to extract this information and append it to the unique gene matrix?

The goal is to use this information as part of DEqMS which requires a precursor/peptide count table and intensities.

**Comment by vdemichev** [comment]
Hi,

This is once of the very often requested features, it's in the todo list.

To do this in R, we recommend using the main report (in .parquet format) and the table() function to count the precursors per gene.

Best,
Vadim

**Reply by abadgerw** [reply]
Thanks, @vdemichev! Do you happen to have set lines of code on hand that I could potentially utilize for this as I'm quite new in R?

**Reply by abadgerw** [reply]
I did some digging. 

Would you recommend the following:

1. Read the parquet file with the arrow package. 
2. Run the table command for the gene column in the parquet file. 
3. Extract the resulting counts and prune so that only the genes found in the unique gene file are represented.

**Reply by vdemichev** [reply, marked as answer]
Something like this will work:

```
library(arrow)
df <- read_parquet('report.parquet')
df <- df[df$Proteotypic == 1 & df$Global.Q.Value <= 0.01,]
df <- df[!duplicated(paste0(df$Genes,df$Stripped.Sequence)),]
t <- table(df$Genes)

mat <- read_tsv_arrow('report.unique_genes_matrix.tsv')
mat<-cbind(t[match(mat$Genes,names(t))],mat)
colnames(mat)[1] = "N.peptides"
```

**Reply by abadgerw** [reply]
Thank you so much!

**Reply by abadgerw** [reply]
@vdemichev I noticed that one of the genes found in the unique gene matrix was not found in the .parquet file when I processed it as you had mentioned to extract precursor counts. Is there a potential reason for this? I did notice one row had no gene name when running the R script so maybe that is the missing gene?

**Reply by vdemichev** [reply]
That's possible, as above I suggested a bit conservative filtering, i.e. you can use any other filtering as in the example above.
